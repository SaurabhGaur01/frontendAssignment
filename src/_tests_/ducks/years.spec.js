import {
    setYears,
    yearsReducer,
} from '../../ducks/years';

const mockKey = 'mockValue';

describe('years Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setYears should match the snapshot' , () => {
            expect(setYears(mockKey)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(yearsReducer()).toEqual(10);
        });
        it('should return the passed state if called with no action' , () => {
            expect(yearsReducer(mockKey)).toEqual(mockKey);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(yearsReducer(mockKey, unknownAction)).toEqual(mockKey);
        });
        it('should return the value set' , () => {
            expect(yearsReducer('', setYears(mockKey))).toEqual(mockKey);
        });
    });
});