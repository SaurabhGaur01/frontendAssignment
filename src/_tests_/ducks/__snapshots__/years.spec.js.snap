// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`years Duck Tests Action Generator setYears should match the snapshot 1`] = `
Object {
  "type": "modules/app/SET_YEARS",
  "value": "mockValue",
}
`;
